torchvision/datasets/video_utils.py

```diff
@@ -0,0 +1,153 @@
+import bisect
+import torch
+import torch.utils.data
+from torchvision.io import read_video_timestamps, read_video
+
+
+def unfold(tensor, size, step, dilation):
+    """
+    similar to tensor.unfold, but with the dilation
+    and specialized for 1d tensors
+
+    Returns all consecutive windows of `size` elements, with
+    `step` between windows. The distance between each element
+    in a window is given by `dilation`.
+    """
+    assert tensor.dim() == 1
+    o_stride = tensor.stride(0)
+    numel = tensor.numel()
+    new_stride = (step * o_stride, dilation * o_stride)
+    new_size = ((numel - (dilation * (size - 1) + 1)) // step + 1, size)
+    if new_size[0] < 1:
+        new_size = (0, size)
+    return torch.as_strided(tensor, new_size, new_stride)
+
+
+class VideoClips(object):
+    """
+    Given a list of video files, computes all consecutive subvideos of size
+    `clip_length_in_frames`, where the distance between each subvideo in the
+    same video is defined by `frames_between_clips`.
+
+    Creating this instance the first time is time-consuming, as it needs to
+    decode all the videos in `video_paths`. It is recommended that you
+    cache the results after instantiation of the class.
+
+    Recreating the clips for different clip lengths is fast, and can be done
+    with the `compute_clips` method.
+    """
+    def __init__(self, video_paths, clip_length_in_frames=16, frames_between_clips=1):
```

Comment:
why not exposing argument `dilation` in the __init__ function, and pass it into `compute_clips` method?

Reply:
Good question. I didn't want to expose it because the `dilation` was a hacky parameter to specify in a general `VideoClips`, because it assumed that all the videos had the same original frame rate, and that they are an integer value.

I've updated the PR with a (hopefully) more robust approach, where I perform an interpolation of the indices in the clip, so that we can support arbitrary conversion from `input_fps` to `target_fps`.

Reply:
I'd think it makes sense to expose it and set it to a default value; there is no harm in general and there are a few applications that don't like interpolation as it can introduce artifacts, and people tend to use it regardless of the actual fps :)

Reply:
But it is conceptually wrong to just step every few frames if the user wants to reduce the fps in general, as different videos might have different fps.

Also note that the approach I'm following is as efficient as the `dilation` one if `step` is integer.

Reply:
Actually, I take that back as the new resampling is doing the same thing `dilation` is doing if the step is an integer